x/auth/simulation/genesis.go

```diff
@@ -0,0 +1,82 @@
+package simulation
+
+// DONTCOVER
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+)
+
+// Simulation parameter constants
+const (
+	MaxMemoChars           = "max_memo_characters"
+	TxSigLimit             = "tx_sig_limit"
+	TxSizeCostPerByte      = "tx_size_cost_per_byte"
+	SigVerifyCostED25519   = "sig_verify_cost_ed25519"
+	SigVerifyCostSECP256K1 = "sig_verify_cost_secp256k1"
+)
+
+// GenMaxMemoChars randomized MaxMemoChars
+func GenMaxMemoChars(r *rand.Rand) uint64 {
+	return uint64(simulation.RandIntBetween(r, 100, 200))
+}
+
+// GenTxSigLimit randomized TxSigLimit
+func GenTxSigLimit(r *rand.Rand) uint64 {
+	return uint64(r.Intn(7) + 1)
+}
+
+// GenTxSizeCostPerByte randomized TxSizeCostPerByte
+func GenTxSizeCostPerByte(r *rand.Rand) uint64 {
+	return uint64(simulation.RandIntBetween(r, 5, 15))
+}
+
+// GenSigVerifyCostED25519 randomized SigVerifyCostED25519
+func GenSigVerifyCostED25519(r *rand.Rand) uint64 {
+	return uint64(simulation.RandIntBetween(r, 500, 1000))
+}
+
+// GenSigVerifyCostSECP256K1 randomized SigVerifyCostSECP256K1
+func GenSigVerifyCostSECP256K1(r *rand.Rand) uint64 {
+	return uint64(simulation.RandIntBetween(r, 500, 1000))
+}
+
+// RandomizedGenState generates a random GenesisState for auth
+func RandomizedGenState(input *module.GeneratorInput) {
```

Comment:
this was to update make the slashing's `MaxEvidence` to use the staking `UnbondingTime` without needing to return any value. The rand is not related to this decision

Reply:
@rigelrozanski can you explain your reasoning for why you think a reference would lead to "spaghetti code"? I believe the name `GeneratorInput` is a misnomer and perhaps thats what led you to state that. This `GeneratorInput` actually contains state (e.g. rand and app state).

Reply:
@alexanderbez yeah, so like originally (many incarnations ago) the sdk had _a ton_ of passing by reference which obfuscated where changes to variables were actually being made, which ended up leading to wild bug chases/headaches. Below is a simple example. In the obfuscated code it's not really clear who is modifying the input, and which fields or who simply using the input - this all become really obvious if you don't pass by reference. 

``` golang
// obfuscated
var input SomeStruct = ...
SomeFunc1(*input)
SomeFunc2(*input)
SomeFunc3(*input)
SomeFunc4(*input)

// crystal-clear
var input SomeStruct = ...
input1.Field1 = SomeFunc1(input)
SomeFunc2(input) // no changes made
input1.Field3 = SomeFunc3(*input)
SomeFunc4(input) // no changes made
```

Now I'm not saying this is what we're doing here, and I'm not saying there aren't good times to pass by reference, I'm just saying if it can be avoided, it will down the road likely improve the code's comprehensibility/maintainability. 

____________

btw what's the current thinking? change the name of `GeneratorInput` eradicate it?